### ISSUE TITLE
In-platform messaging

### ISSUE THREAD

**Issue by kaylarep** [opencollective/opencollective#4699]
Would be great to have a 'Messages' section in the platform
- record of past conversations to avoid duplicating messages
- can see that the person has already been contacted
- can view message history all in one place- no need to hunt thru emails (find who was even emailed)

User story:
I am very wary to send messages using the platform, instead preferring to move into email, because I will have no record of having sent a message, and might needlessly duplicate communications. It's confusing and sometimes frustrating (because I also do not have access to the person's email contact) to use these workarounds to communicate with users.

At the very least having a button to 'Send a copy to myself' at the bottom would be a good start.

**Comment by Betree** [comment]
That's not an easy one to build and maintain, but I 100% agree there. Having an internal messaging system would help in many different cases:
- **Better contact form**: An internal messaging system would prevent issues like https://github.com/opencollective/opencollective/issues/3106 and would allow for a shared message history for collectives/organizations - along with a way to know whether another admin already opened/replied to the message. Would also resolve the need for https://github.com/opencollective/opencollective/issues/2840 and https://github.com/opencollective/opencollective/issues/2263.
- **For collective applications**, we know for a fact that collectives and fiscal hosts often need to discuss the approval. Today we push for Conversations (when the conv. can be public) or email, which is sad because we lose the discussion history on Open Collective (though we could resolve this one by having a more simple & specific feature: allowing for comments threads like the ones used for expenses, but for collective applications)
- Provide an MVP for collective/admins to **contact users automatically after a contribution** (via the API) to give them some specific details: https://github.com/opencollective/opencollective/issues/3234
- In the context of GDPR (#4537), one difficulty is for collectives/events who need to **contact a single contributor** - for example, a gold sponsor to whom you want to send a unique code. The updates feature does not allow targeting specific tiers at the moment.
- Still in the context of GDPR (#4537), we should anyway favor implementing communication tools rather than sharing emails & personal info with admins.

**Comment by shannondwray** [comment]
This is done by most companies with Intercom - https://www.intercom.com/

**Comment by Betree** [comment]
@shannondwray https://www.intercom.com seems to be more about having a support chat, not allowing users to send messages to each other (e.g. a contributor sends an email to a collective admin at Babel)? Or am I missing something?

**Comment by kaylarep** [comment]
yes I think you're correct @Betree ...Intercom could maybe help the internal OC team with support, but for user-to-user messaging, I'm not sure it will cover that

**Comment by kaylarep** [comment]
if this also included 'templates' or 'canned replies', could potentially eliminate the need for an external messaging platform entirely (e.g. OCF uses Freshdesk & Slack & OC messaging to manage our comms with each other & with hosted groups & their admins. It's a bit all over the place, so would be amazing to have it all one in place.)

e.g, groups submits an application, notification is sent to OCF admin via email. API connects the application to our slack, also the initial email is forwarded to Freshdesk. we discuss the application internally in slack. we respond to the applications in Freshdesk (this way the whole team can view the record of the conversation with the applicant. also FD has canned replies that we have customized to make processing applications more efficient)

**Comment by Upperholme** [comment]
Bump. From the perspective of being a fiscal host, there seems to be no simple way to enter into a dialogue with a collective that has applied to be hosted by us. I've tried sending messages through the web interface, but I have no idea of whether the message has been sent. I've certainly never received a reply.

**Comment by Upperholme** [comment]
At the very least, with pending applications, I should be able to see the contact email address/es for the applicant collective.

**Comment by BenJam** [comment]
I'd like to use this issue to scope many of the requests in https://github.com/opencollective/opencollective/issues/5330 around messaging inc.  #3357  and #5382

**Comment by kaylarep** [comment]
As the team has been discussing focusing more on what we already do as opposed to building out new features, I'm imagining in-platform messaging could get quite complex and heavy...

Is there an integration/API we could build instead? OC team uses Slack, but ideally, it would be an OSS, or at least something with greater access than Slack (many of OCF's hosted groups would like to be able use Slack, but arent because of TechSoup.)

Or is there a way to convert our Conversations feature into some sort of internal messaging?

Alternatitvely, as a very lightweight MVP: when someone clicks the 'contact' button, the module popup would say something like "This admin prefers to be contacted via ___" and there would also be something in the collective settings of "Preferred contact information".

As it stand right now, (as a fiscal host admin receiving messages via the platform), the email without any context/thread is pretty unworkable for me. All comms I do with collectives is either in freshdesk or slack, barely even email anymore, because my email is too flooded (https://github.com/opencollective/opencollective/issues/3357)

**Comment by Betree** [comment]
> As the team has been discussing focusing more on what we already do as opposed to building out new features, I'm imagining in-platform messaging could get quite complex and heavy...

I wouldn't oppose the two. In my view, in-platform messaging is a way to improve what we do today, not a way to conquer new users or new markets. 

> Is there an integration/API we could build instead? OC team uses Slack, but ideally, it would be an OSS, or at least something with greater access than Slack (many of OCF's hosted groups would like to be able use Slack, but arent because of TechSoup.)

There are 3rd party APIs that we can use to build this functionality. It's worth investigating, but my gut feeling tells me that 1. It won't simplify the implementation that much 2. It will make us dependent on a 3rd party tool we have little control on 3. I wouldn't want to store things as sensitive as private messages outside of Open Collective.

> Or is there a way to convert our Conversations feature into some sort of internal messaging?

We could re-use some parts of it (e.g. the comment form, the emoji reactions, etc). This issue would still be a large project that will require a big investment from design and engineering. It's a complex one that will impact multiple parts of the user flows, I don't think there are magic shortcuts to make it happen outside of a proper prioritization (_which I would personally strongly support, in my opinion this issue along with https://github.com/opencollective/opencollective/issues/5382 have a massive potential to improve the UX_)

> Alternatitvely, as a very lightweight MVP: when someone clicks the 'contact' button, the module popup would say something like "This admin prefers to be contacted via ___" and there would also be something in the collective settings of "Preferred contact information".

This has been proposed in https://github.com/opencollective/opencollective/issues/2263

**Comment by kaylarep** [comment]
thanks for these replies and your perspective @Betree ! if you think its possible to make this, then would definitely be a great improvement to the UX.

my main thought is I would hate to have a clunky messaging system like what my banks/billing companies use. its a pain to have to manage their in-platform messaging along with my personal email/work email, all the other messaging platforms/portals we use today... especially since we don't have an app...

user story -pain point- : my electricity provider sends me an email saying "you have a new message. please check in our platform". I'm not near my computer so I access their website on my phone (they also don't have an app).  I don't have the password saved on my phone, have to reset it/find it etc.  finally get into their platform and the message says something mundane, obvious, and pretty unnecessary; and it's a very frustrating experience to engage with + happens too frequently with these types of messages coming frequently/randomly from the bank, utility services, healthcare portals, government portals, etc... its annoying and too much to manage; so I definitely would not want OC to feel like one of those!

**Comment by kaylarep** [comment]
I just replied to a comment/thread in an expense (as a fiscal host admin) for the first time, and that format seems very clear and simple! If that can be adapted for an in-platform messaging system, that might be the way to do it.  

or even using that same format for pending applications would be a great place to start (having canned replies for pending applications would be vital too. otherwise i would still be unlikely to use it).

**Comment by BenJam** [comment]
Just pulling together a plan for #5501 and this:

> user story -pain point- : my electricity provider sends me an email saying "you have a new message. please check in our platform

is the big anti-pattern for me. Yes it might be privacy-preserving, but at what cost to the user and the service itself. If the content of this message was 'we need some more details from you to pay you money for your groceries' it's too important to obfusticate behind some generic notice.

**Comment by shannondwray** [comment]
Feature requested through support for a collective admin to be able to message a donor thanking them for their contribution, I think implementing in platform messaging will make people feel more connected to the communities they are giving to

**Comment by kaylarep** [comment]
I just want to add that even though I created the issue, I'm not convinced this is a good idea. It seems like it could be quite a complex thing to create an in-platform messaging system, and might not be a feature we want to focus efforts on.

Instead, I think we can consider the root problem we are trying to solve for: how to improve communications between collectives, admins, hosts, community members. 

And maybe we can brainstorm a more lightweight solution?

**Comment by shannondwray** [comment]
This would also be useful for following up with people who have contributed manually via bank transfer
